### PR TITLE
Fixed story button saving & more improvements

### PR DESCRIPTION
**Pull request by Andrerm124**
- Fixed the story button saving
- Fixed Discover UI blocking
- Moved Lenses.db to Snapprefs folder

### CODE REVIEW

File: app/src/main/java/com/marz/snapprefs/Stories.java [hunk 1]
```diff
@@ -97,6 +100,9 @@ protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
                 }
             }
         });
+
+        if( Preferences.getBool(Prefs.DISCOVER_UI))
+            findAndHookMethod("atk", lpparam.classLoader, "c", List.class, XC_MethodReplacement.returnConstant(new ArrayList<>()));
```
Comment (marzika) [inline app/src/main/java/com/marz/snapprefs/Stories.java]: Can you please move it to the Obfuscator class?

File: app/src/main/java/com/marz/snapprefs/Saving.java [hunk 2]
```diff
@@ -138,7 +136,15 @@ protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
                             super.beforeHookedMethod(param);
                             Log.d("snapprefs", "### START StoryViewerMediaCache ###");
                             Object godPacket = param.args[1];
-                            Object storySnap = callMethod(godPacket, Obfuscator.save.SDP_GET_OBJECT, "STORY_REPLY_SNAP");
+                            Object c = getObjectField(param.thisObject, "c");
+                            String POSTER_USERNAME = (String) callMethod(godPacket, Obfuscator.save.SDP_GET_STRING, "POSTER_USERNAME");
+                            String CLIENT_ID = (String) callMethod(godPacket, Obfuscator.save.SDP_GET_STRING, "CLIENT_ID");
+                            Object storySnap = callMethod(c, "a", POSTER_USERNAME, CLIENT_ID);
```
Comment (marzika) [inline app/src/main/java/com/marz/snapprefs/Saving.java]: same for L139-142

**Comment by Andrerm124** [comment]
Will do!

File: app/src/main/java/com/marz/snapprefs/HookedLayouts.java [hunk 3]
```diff
@@ -163,6 +164,7 @@ public void handleLayoutInflated(LayoutInflatedParam liparam) throws Throwable {
     public static void addShareIcon(final XC_InitPackageResources.InitPackageResourcesParam resparam) {
         resparam.res.hookLayout(Common.PACKAGE_SNAP, "layout", "camera_preview", new XC_LayoutInflated() {
             public void handleLayoutInflated(LayoutInflatedParam liparam) throws Throwable {
+                Logger.log("Doing shit?");
                 final RelativeLayout relativeLayout =
```
Comment (marzika) [inline app/src/main/java/com/marz/snapprefs/HookedLayouts.java:166]: Also, can you please remove the log here?

**Review by marzika — CHANGES_REQUESTED** [review]
commented them

**Comment by Andrerm124** [comment]
@marzika Made those changes

**Pull request merged**